cardano-config/src/Cardano/TracingOrphanInstances/HardFork.hs

```diff
@@ -0,0 +1,266 @@
+import           Cardano.TracingOrphanInstances.Common
+import           Cardano.TracingOrphanInstances.Consensus ()
+
+import           Ouroboros.Consensus.Block (BlockProtocol, ForgeState)
+import           Ouroboros.Consensus.Protocol.Abstract
+                   (ValidationErr, CannotLead)
+import           Ouroboros.Consensus.Ledger.Abstract (LedgerError)
+import           Ouroboros.Consensus.Ledger.SupportsMempool
+                   (GenTx, TxId, ApplyTxErr)
+import           Ouroboros.Consensus.HeaderValidation
+                   (OtherHeaderEnvelopeError)
+import           Ouroboros.Consensus.HardFork.Combinator
+import           Ouroboros.Consensus.HardFork.Combinator.AcrossEras
+                   (OneEraValidationErr(..), OneEraLedgerError(..),
+                    OneEraEnvelopeErr(..), OneEraCannotLead(..),
+                    EraMismatch(..), mkEraMismatch)
+import           Ouroboros.Consensus.TypeFamilyWrappers
+
+import           Ouroboros.Consensus.Util.Condense (Condense(..))
+
+
+--
+-- instances for hashes
+--
+
+instance Condense (OneEraHash xs) where
+    condense = condense . Crypto.UnsafeHash . getOneEraHash
+
+
+--
+-- instances for Header HardForkBlock
+--
+
+instance All (Condense `Compose` Header) xs => Condense (Header (HardForkBlock xs)) where
+    condense =
+          hcollapse
+        . hcmap (Proxy @ (Condense `Compose` Header)) (K . condense)
+        . getOneEraHeader
+        . getHardForkHeader
+
+instance All (ToObject `Compose` Header) xs => ToObject (Header (HardForkBlock xs)) where
+    toObject verb =
+          hcollapse
+        . hcmap (Proxy @ (ToObject `Compose` Header)) (K . toObject verb)
+        . getOneEraHeader
+        . getHardForkHeader
+
+
+--
+-- instances for GenTx HardForkBlock
+--
+
+instance Condense (GenTx (HardForkBlock xs)) where
+  condense = error "TODO: replace with versions from upstream"
+
+instance Condense (TxId (GenTx (HardForkBlock xs))) where
+  condense = error "TODO: replace with versions from upstream"
+
+instance All (Compose ToObject GenTx) xs => ToObject (GenTx (HardForkBlock xs)) where
+    toObject verb =
+          hcollapse
+        . hcmap (Proxy @ (ToObject `Compose` GenTx)) (K . toObject verb)
+        . getOneEraGenTx
+        . getHardForkGenTx
+
+instance  All (Compose ToJSON WrapGenTxId) xs => ToJSON (TxId (GenTx (HardForkBlock xs))) where
+    toJSON =
+          hcollapse
+        . hcmap (Proxy @ (ToJSON `Compose` WrapGenTxId)) (K . toJSON)
+        . getOneEraGenTxId
+        . getHardForkGenTxId
+
+instance ToJSON (TxId (GenTx blk)) => ToJSON (WrapGenTxId blk) where
+    toJSON = toJSON . unwrapGenTxId
+
+
+--
+-- instances for GenTx HardForkBlock
+--
+
+{-
+instance HasKESMetricsData (HardForkBlock xs) where
+    getKESMetricsData :: ProtocolInfo m (HardForkBlock xs)
+                      -> ForgeState (HardForkBlock xs)
+                      -> KESMetricsData
+    getKESMetricsData protoInfo forgeState =
+-}
```

Comment:
@mrBliss I could do with help on this one. For the `ForgeState` it's a product, but each one will expect a `ProtocolInfo` projected to it's individual block type.

Reply:
Why does it need the whole `ProtocolInfo`? Writing this projection (let alone generically) will be impossible: for one, the `ProtocolInfo` contains the initial ledger state, but the `ProtocolInfo` for a `HardForkBlock` will only contain the initial ledger state for the first era, *not for all of them*.

A [quick search](https://github.com/input-output-hk/cardano-node/search?q=getKESMetricsData&unscoped_q=getKESMetricsData) revealed that the `ProtocolInfo` isn't actually used, so I would just remove it.

Reply:
Oh, nice. Ok, thanks.

Reply:
Turns out this is a bit harder. We can solve in a follow-up PR once more HFC stuff is merged in consensus.